### DISCUSSION TITLE
Implementing 301 Redirects with Status Code Control in Next.js 13 Server Components (TypeScript)

### DISCUSSION THREAD

**Discussion by devAndDes** [vercel/next.js#67242] [Help]
### Summary

I'm working with Next.js 13 (TypeScript) and I'm encountering challenges in managing redirects effectively. My application dynamically fetches destination URLs from a CRM, allowing users to add, remove, and modify them.

I've experimented with various approaches, but haven't yet achieved the desired behavior. Specifically, while the redirect hook in server components triggers the redirect, it doesn't send the crucial 301 status code.

Here's an example of what I've tried:

return {
  redirect: {
    source: selectedPage.url,
    destination: selectedPage.redirect,
    permanent: true, // 301 redirect
  },
};

Unfortunately, the status code doesn't change to 301.

To summarize:
How can I use redirect from server component with status 301 ?
